snap/local/hooks/cmd/install/install.go

```diff
@@ -389,22 +426,48 @@ func main() {
 		os.Exit(1)
 	}
 
-	// just like the configure hook, this code needs to iterate over every service
-	// and setup .env files for each if required...
-	for _, v := range hooks.Services {
-		serviceCfg := hooks.EnvConfig + "." + v
-		envJSON, err := cli.Config(serviceCfg)
-		if err != nil {
-			hooks.Error(fmt.Sprintf("edgexfoundry:install failed to read service %s configuration - %v", v, err))
+	// Stop and disable all *enabled* services as they will be
+	// re-enabled in the configure hook if the config option
+	// 'install=true'.
+	for _, s := range enabledServices {
+		if err = cli.Stop(s, true); err != nil {
+			hooks.Error(fmt.Sprintf("edgexfoundry:install can't disable: %s; %v", s, err))
 			os.Exit(1)
 		}
 
-		if envJSON != "" {
-			hooks.Debug(fmt.Sprintf("edgexfoundry:install: service envJSON: %s", envJSON))
-			if err := hooks.HandleEdgeXConfig(v, envJSON, nil); err != nil {
-				hooks.Error(fmt.Sprintf("edgexfoundry:install failed to process service %s configuration - %v", v, err))
-				os.Exit(1)
-			}
+		hooks.Info(fmt.Sprintf("edgexfoundry:install disabling service: %s", s))
+
+		// NOTE - snap service names don't match service keys
+		switch s {
+
+		case "security-proxy-setup":
+			s = "security-proxy"
+		case "security-secretstore-setup":
+			s = "security-secret-store"
+
+		// these will be started in the install hook if "security-secret-store=install"
+		// as such, we don't set config service keys for any of them
+		case "security-consul-bootstrapper", "security-bootstrapper-redis", "vault":
+			continue
+
+		// these will be started in the install hook if "security-proxy=install"
+		// as such, we don't set config service keys for any of them
+		case "kong-daemon", "postgres":
+			continue
+		}
+
+		// Use "install" instead of "on"; this allows the configure hook to unset
+		// all of the service options set to "install", and leave configuration
+		// that actually happened as a result of provided config (e.g. from the
+		// gadget) alone
+		if err = cli.SetConfig(s, "install"); err != nil {
+			hooks.Error(fmt.Sprintf("edgexfoundry:install setting service: %s to 'on'; %v", s, err))
+			os.Exit(1)
 		}
 	}
+
+	if err = cli.SetConfig("install", "true"); err != nil {
```

Comment:
it does seem inconsistent to me in hindsight that snapd let's you set config items in the install hook (before the configure hook runs), but it doesn't let you read them before the configure hook runs :-|